plugins/framework/tests/test_accel_patcher.py

```diff
@@ -0,0 +1,151 @@
+# Copyright The IBM Tuning Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Third Party
+from accelerate import Accelerator
+import pytest  # pylint: disable=import-error
+import torch
+
+# First Party
+from fms_acceleration.accelerator_patcher import (
+    AcceleratorPatcher,
+    AcceleratorPatcherComponent,
+    AcceleratorRuleReplace,
+)
+from fms_acceleration.utils.test_utils import instantiate_accel_patcher
+
+RULE_ID = "test"
+
+
+def test_AP_rule_raises_correct_errors():
+    # not specifying any replacement objects will throw an error
+    with pytest.raises(
+        AssertionError,
+        match="either replacement or replacement should be specified",
+    ):
+        AcceleratorRuleReplace(
+            rule_id=RULE_ID,
+            component=AcceleratorPatcherComponent.data_loader,
+            replacement=None,
+            replacement_builder=None,
+        )
+
+    # Ensure that rule registration throws error when attempting
+    # to specify an unknown flag for an unsupported behaviour
+    # handling of the component by AP
+    with pytest.raises(
+        AssertionError,
+        match="Invalid special behavior kwargs in",
```

Comment:
if the message also provides the key `"unsupported_kwarg"` inside, and its easy to specify, then maybe should include